src/transformStateWithClones.js

```diff
@@ -8,6 +8,35 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateCopy = Object.assign({}, state);
+  const arrayOfOriginals = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
```

Comment:
the perfect situation to use `switch` block - do it

Reply:
please review new commit, thank you!